src/librustc_privacy/lib.rs

```diff
@@ -206,7 +208,7 @@ impl<'a, 'tcx, V> TypeVisitor<'tcx> for DefIdVisitorSkeleton<'_, 'a, 'tcx, V>
                 bug!("unexpected type: {:?}", ty),
```

Comment:
There are a few other cases where `visit_ty` can recurse, you need to avoid them.